### PR TITLE
Add support for node lookups

### PR DESCRIPTION
**Pull request by clpe04**

### CODE REVIEW

File: src/neo4clj/query_builder.clj [hunk 1]
```diff
@@ -162,13 +167,27 @@
...
+  [{:keys [ref-id] :as entity}]
+  (cond
+    (nil? entity) nil
+    (string? entity) entity
+    (nil? ref-id) (assoc entity :ref-id (cypher/gen-ref-id))
+    :else entity))
+
+(defn ensure-rel-nodes-have-ref-id
+  [{:keys [from to] :as rel}]
+  (update (update rel :from ensure-ref-id) :to ensure-ref-id))
```
Comment (jacobemcken) [inline src/neo4clj/query_builder.clj]: I think the following is easier to read:
```clojure
(-> rel
    (update :from ensure-ref-id)
    (update :to ensure-ref-id))

File: src/neo4clj/query_builder.clj [hunk 2]
```diff
@@ -162,13 +167,27 @@
...
+(defn- ensure-ref-id
+  [{:keys [ref-id] :as entity}]
+  (cond
+    (nil? entity) nil
+    (string? entity) entity
+    (nil? ref-id) (assoc entity :ref-id (cypher/gen-ref-id))
+    :else entity))
+
+(defn ensure-rel-nodes-have-ref-id
+  [{:keys [from to] :as rel}]
```
Comment (jacobemcken) [inline src/neo4clj/query_builder.clj]: No need to destructure `from` and `to`.

File: test/neo4clj/query_builder_test.clj [hunk 3]
```diff
@@ -110,55 +137,80 @@
...
-      [{:ref-id "r1" :type :owner :from "n1" :to {:ref-id "n2" :labels [:person] :id 4}}
-       {:ref-id "r2" :type :employee :from "n1" :to "n3" :exists false}]
-
-      "MATCH (n1:Person) MATCH (n3:Company) WHERE ID(n3) = 6 AND NOT (n1)-[:EMPLOYEE]->(n3)"
-      {"n1" {:ref-id "n1" :labels [:person]}
-       "n3" {:ref-id "n3" :labels [:company]:id 6}}
-      [{:ref-id "r2" :type :employee :from "n1" :to "n3" :exists false}]
-      )))
+  (let [next-gensym (atom 0)]
+    (with-redefs [cypher/gen-ref-id (fn [] (str "G__" (swap! next-gensym inc)))]
```
Comment (jacobemcken) [inline test/neo4clj/query_builder_test.clj:162]: I would put the `with-redefs` inside the `are` to reset the atom for every test. When used in mulitiple test cases you can re-arrange the data sets without affecting the tests. Imagine wanting the test case using `G__2` to be placed before the one using `G__1`... that would require also touching the data.

File: docs/examples.md [hunk 4]
```diff
@@ -136,6 +136,18 @@ To learn more about the Clojure representation of a node please see our [represe
                                    :last-name "Anderson"}})
 ~~~
 
+### Find a single node in Neo4j
+
+The entry given to `find-node!` is a Node Lookup representation. To learn more about the Clojure Node Lookup representation see our [representations](representations.md) page
```
Comment (jacobemcken) [inline docs/examples.md]: Now the three first paragraphs repeat (see below):
> ... To learn more about the Clojure Node Lookup representation see ...

Is there a better way to turn attention to the representations without repeating it?

**Review by jacobemcken — APPROVED** [review]

File: src/neo4clj/query_builder.clj (hunk 1 above)
Comment (clpe04) [reply src/neo4clj/query_builder.clj]: Fixed

File: src/neo4clj/query_builder.clj (hunk 2 above)
Comment (clpe04) [reply src/neo4clj/query_builder.clj]: Fixed

File: test/neo4clj/query_builder_test.clj (hunk 3 above)
Comment (clpe04) [reply test/neo4clj/query_builder_test.clj:162]: Ignored

**Pull request merged**